controller/state/statedb.go

```diff
@@ -144,6 +148,19 @@ func newStateDBWithNotify(ctx context.Context, dbConn DBConnector, migrator Migr
 		log.Infow("recovered scheduled tasks", "task_count", count)
 	}
 
+	storeLS := storeLS(st.Store(context.Background()))
```

Comment:
part of me would suggest exposing the datastore here, but having instantiation of libp2p and legs be elsewhere. seems like an odd mixing of concerns to setup a whole network transport stack inside of a database instantiation.

Reply:
I could be wrong but I don't think I actually see any real common deps other than the priv key?

Reply:
it's a good point. i just don't know where i actually want to make it the legs/libp2p host :)